### PR TITLE
ui: add filter support for metadata fields

### PR DESCRIPTION
**Pull request by masco**

### CODE REVIEW

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

File: statics/js/components/topology.js [hunk 1]
```diff
@@ -626,6 +630,11 @@ var TopologyComponent = {
       this.addFilterIstio(control, label, this.gremlinK8sTypes(types));
     },
 
+    setMetadataHiddenFields: function() {
+      let fields = app.getConfigValue('topology.metadata.hidden_fields')
+      this.hiddenFields = fields
```
Comment (hunchback) [inline statics/js/components/topology.js]: Protect against missing or no configuration:
```
let key = 'topology.metadata.hidden_fields'
let val = app.getConfigValue(key)
if (! Array.isArray(val)) {
  log.console('Wrong config ' + key + ' go value: ' + val)
  return
}
this.hiddenFields = val
```

File: statics/js/components/topology.js [hunk 2]
```diff
@@ -626,6 +630,11 @@ var TopologyComponent = {
       this.addFilterIstio(control, label, this.gremlinK8sTypes(types));
     },
 
+    setMetadataHiddenFields: function() {
```
Comment (hunchback) [inline statics/js/components/topology.js]: s/setMetadataHiddenFields/setHiddenFields/

File: statics/js/components/object-detail.js [hunk 3]
```diff
@@ -83,12 +93,38 @@ Vue.component('object-detail', {
...
+              </a>\
+            </li>\
+          </button-dropdown>\
       </div>\
     </div>\
   ',
 
-  methods: {
+  data: function() {
+    return{
```
Comment (hunchback) [inline statics/js/components/object-detail.js]: s/return{/return {/

File: statics/css/skydive.css [hunk 4]
```diff
@@ -508,20 +508,29 @@ circle.manager {
   margin-top: 0px;
 }
 
-.dynamic-table .fa-cog, .dynamic-table .btn, .dynamic-table .btn[active], .dynamic-table .btn[focus] {
+.dynamic-table .fa-cog, .dynamic-table .btn, .dynamic-table .btn[active], .dynamic-table .btn[focus],
+.object-detail-filter .fa-cog, .object-detail-filter .btn, .object-detail-filter .btn[active],
```
Comment (hunchback) [inline statics/css/skydive.css]: you could alternatively leave this block as is and then append the `dynamic-table` class as a secondary class such in:

```
 <div v-if="filterIcon" class="object-detail-filter">\
 ...
```

Would change to:

```
 <div v-if="filterIcon" class="object-detail-filter dynamic-table">\
  ...
```

**Comment by hunchback** [comment]
when using an old skydive configuration file, pressing on nodes does not display any Metadata:

image

**Comment by masco-enovance** [comment]
@hunchback it is working in chrome for me. look like your chrome browser is not connected with analyzer. check the icon near logout button.

File: statics/js/components/topology.js (hunk 1 above)
Comment (masco-enovance) [reply statics/js/components/topology.js]: fixed

File: statics/js/components/topology.js (hunk 2 above)
Comment (masco-enovance) [reply statics/js/components/topology.js]: done

File: statics/js/components/object-detail.js (hunk 3 above)
Comment (masco-enovance) [reply statics/js/components/object-detail.js]: done

File: statics/css/skydive.css (hunk 4 above)
Comment (masco-enovance) [reply statics/css/skydive.css]: done

**Comment by masco-enovance** [comment]
run skydive-k8s-tests

**Comment by lebauce** [comment]
Implemented in new UI

**Pull request closed without merging**